tests/app/questionnaire/test_date_rules.py

```diff
@@ -3,153 +3,149 @@
-        self.assertTrue(result)
-
-        answer_value = "2018-02-03"
-        result = evaluate_date_rule(
-            when, answer_store, get_schema_mock(), None, answer_value
-        )
-        self.assertFalse(result)
-
-    def test_evaluate_date_rule_invalid(self):
-        when = {
-            "id": "date-answer",
-            "condition": "greater than",
-            "date_comparison": {"id": "compare_date_answer"},
-        }
-        answer_store = AnswerStore({})
-        result = evaluate_date_rule(when, answer_store, get_schema_mock(), None, None)
-
-        self.assertFalse(result)
-
-    def test_do_not_go_to_next_question_for_date_answer(self):
-        goto_rule = {
-            "id": "next-question",
-            "when": [
-                {
-                    "id": "date-answer",
-                    "condition": "equals",
-                    "date_comparison": {"value": "2018-01"},
-                }
-            ],
-        }
-
-        answer_store = AnswerStore({})
-        answer_store.add_or_update(Answer(answer_id="date-answer", value="2018-02-01"))
-
-        current_location = Location(section_id="some-section", block_id="some-block")
-
-        self.assertFalse(
-            evaluate_goto(
-                goto_rule=goto_rule,
-                schema=get_schema_mock(),
-                metadata={},
-                answer_store=answer_store,
-                list_store=ListStore(),
-                current_location=current_location,
-            )
-        )
+
+
+def test_evaluate_date_rule_equals_with_value_now(questionnaire_schema):
+    when = {
+        "id": "date-answer",
+        "condition": "equals",
+        "date_comparison": {"value": "now"},
+    }
+
+    answer_value = datetime.now(tz=timezone.utc).strftime("%Y-%m-%d")
+    result = evaluate_date_rule(when, None, questionnaire_schema, None, answer_value)
+    assert result
+
+    answer_value = "2000-01-01"
+    result = evaluate_date_rule(when, None, questionnaire_schema, None, answer_value)
+    assert not result
+
+
+def test_evaluate_date_rule_equals_with_with_offset(questionnaire_schema):
+    when = {
+        "id": "date-answer",
+        "condition": "equals",
+        "date_comparison": {
+            "value": "2019-03-31",
+            "offset_by": {"days": 1, "months": 1, "years": 1},
+        },
+    }
+
+    answer_value = "2020-05-01"
+    result = evaluate_date_rule(when, None, questionnaire_schema, None, answer_value)
+    assert result
+
+    when = {
+        "id": "date-answer",
+        "condition": "equals",
+        "date_comparison": {
+            "value": "2021-04-01",
+            "offset_by": {"days": -1, "months": -1, "years": -1},
+        },
+    }
+
+    answer_value = "2020-02-29"
+    result = evaluate_date_rule(when, None, questionnaire_schema, None, answer_value)
+    assert result
+
+
+def test_evaluate_date_rule_not_equals_with_value_year_month(questionnaire_schema):
+    when = {
+        "id": "date-answer",
```

Comment:
same as above, can use `parametrize`

Reply:
https://github.com/ONSdigital/eq-questionnaire-runner/pull/799#discussion_r815865189

Reply:
got it, saw the duplication when I looked again, I though ash was referring to the date-id part rather than the whole lot